snap-geometry-edits/src/main/java/com/esri/arcgismaps/sample/snapgeometryedits/components/MapViewModel.kt

```diff
@@ -0,0 +1,255 @@
+            geometryEditor.start(selectedGeometry)
+            toggleButtons()
+        }
+    }
+
+    /**
+     * Stops the GeometryEditor and updates the identified graphic or calls [createGraphic].
+     */
+    fun stopEditor() {
+        if (identifiedGraphic.geometry != null) {
+            identifiedGraphic.geometry = geometryEditor.stop()
+            identifiedGraphic.isSelected = false
+        } else {
+            if (geometryEditor.isStarted.value) {
+                createGraphic()
+            }
+        }
+        toggleButtons()
+    }
+
+    /**
+     * Creates a graphic from the geometry and add it to the GraphicsOverlay.
+     */
+    private fun createGraphic() {
+        val geometry = geometryEditor.stop()
+            ?: return messageDialogVM.showMessageDialog(
+                "Error!",
+                "Error stopping editing session"
+            )
+        val graphic = Graphic(geometry)
+
+        when (geometry) {
+            is Point, is Multipoint -> graphic.symbol = GeometryEditorStyle().vertexSymbol
+            is Polyline -> graphic.symbol = GeometryEditorStyle().lineSymbol
+            is Polygon -> graphic.symbol = GeometryEditorStyle().fillSymbol
+            else -> {}
+        }
+        graphicsOverlay.graphics.add(graphic)
+        graphic.isSelected = false
+    }
+
+    /**
+     * Deletes the selected element and stops the geometry editor if there are no
+     * more elements in the geometry.
+     */
+    fun deleteSelection() {
+        if (geometryEditor.selectedElement.value != null) {
+            geometryEditor.deleteSelectedElement()
+            if (geometryEditor.geometry.value?.isEmpty == true) {
+                geometryEditor.stop()
+                toggleButtons()
+            }
+        }
+    }
+
+    /**
+     * Reverts the last event on the geometry editor.
+     */
+    fun editorUndo() {
+        geometryEditor.undo()
+    }
+
+    /**
+     * Identifies the graphic at the tapped screen coordinate in the provided [singleTapConfirmedEvent]
+     * and starts the GeometryEditor using the identified graphic's geometry. Hide the BottomSheet on
+     * [singleTapConfirmedEvent].
+     */
+    fun identify(singleTapConfirmedEvent: SingleTapConfirmedEvent) {
+        sampleCoroutineScope.launch {
+            val graphicsResult = mapViewProxy.identifyGraphicsOverlays(
+                screenCoordinate = singleTapConfirmedEvent.screenCoordinate,
+                tolerance = 10.0.dp,
+                returnPopupsOnly = false
+            ).getOrNull()
+
+            if (!geometryEditor.isStarted.value) {
+                if (graphicsResult != null) {
+                    if (graphicsResult.isNotEmpty()) {
+                        identifiedGraphic = graphicsResult[0].graphics[0]
+                        identifiedGraphic.isSelected = true
+                        identifiedGraphic.geometry?.let { geometryEditor.start(it) }
+                        toggleButtons()
```

Comment:
nitpick: if this nullability check is done before starting the editor then it should also be done for toggling the buttons or else they could get out of sync
```suggestion
                        identifiedGraphic.geometry?.let {
                            geometryEditor.start(it)
                            toggleButtons()
                        }
```